sabnzbd/utils/file_extension.py

```diff
@@ -0,0 +1,313 @@
+    "asp",
+    "aspx",
+    "avi",
+    "bak",
+    "bat",
+    "bin",
+    "bmp",
+    "c",
+    "cab",
+    "cda",
+    "cer",
+    "cfg",
+    "cfm",
+    "cgi",
+    "cgi",
+    "cgi",
+    "class",
+    "com",
+    "cpl",
+    "cpp",
+    "cs",
+    "css",
+    "csv",
+    "cur",
+    "dat",
+    "db",
+    "dbf",
+    "deb",
+    "dll",
+    "dmg",
+    "dmp",
+    "doc",
+    "docx",
+    "drv",
+    "email",
+    "eml",
+    "emlx",
+    "exe",
+    "flv",
+    "fnt",
+    "fon",
+    "gadget",
+    "gif",
+    "h",
+    "h264",
+    "htm",
+    "html",
+    "icns",
+    "ico",
+    "ico",
+    "ini",
+    "iso",
+    "jar",
+    "java",
+    "jpeg",
+    "jpg",
+    "js",
+    "jsp",
+    "key",
+    "lnk",
+    "log",
+    "m4v",
+    "mdb",
+    "mid",
+    "midi",
+    "mkv",
+    "mov",
+    "mp3",
+    "mp4",
+    "mpa",
+    "mpeg",
+    "mpg",
+    "msg",
+    "msi",
+    "msi",
+    "odp",
+    "ods",
+    "odt",
+    "oft",
+    "ogg",
+    "ost",
+    "otf",
+    "part",
+    "pdf",
+    "php",
+    "php",
+    "pkg",
+    "pl",
+    "pl",
+    "pl",
+    "png",
+    "pps",
+    "ppt",
+    "pptx",
+    "ps",
+    "psd",
+    "pst",
+    "py",
+    "py",
+    "py",
+    "rar",
+    "rm",
+    "rpm",
+    "rss",
+    "rtf",
+    "sav",
+    "sh",
+    "sql",
+    "svg",
+    "swf",
+    "swift",
+    "sys",
+    "tar",
+    "tar",
+    "gz",
+    "tex",
+    "tif",
+    "tiff",
+    "tmp",
+    "toast",
+    "ttf",
+    "txt",
+    "vb",
+    "vcd",
+    "vcf",
+    "vob",
+    "wav",
+    "wma",
+    "wmv",
+    "wpd",
+    "wpl",
+    "wsf",
+    "xhtml",
+    "xls",
+    "xlsm",
+    "xlsx",
+    "xml",
+    "z",
+    "zip",
+)
+
+DOWNLOAD_EXT = (
+    "ass",
+    "avi",
+    "bat",
+    "bdmv",
+    "bin",
+    "bup",
+    "clpi",
+    "crx",
+    "db",
+    "diz",
+    "djvu",
+    "docx",
+    "epub",
+    "exe",
+    "flac",
+    "gif",
+    "gz",
+    "htm",
+    "html",
+    "icns",
+    "ico",
+    "idx",
+    "ifo",
+    "img",
+    "inf",
+    "info",
+    "ini",
+    "iso",
+    "jpg",
+    "log",
+    "m2ts",
+    "m3u",
+    "m4a",
+    "mkv",
+    "mp3",
+    "mp4",
+    "mpls",
+    "mx",
+    "nfo",
+    "nib",
+    "nzb",
+    "otf",
+    "par2",
+    "part",
+    "pdf",
+    "pem",
+    "php",
+    "plist",
+    "png",
+    "py",
+    "rar",
+    "releaseinfo",
+    "rev",
+    "sfv",
+    "sh",
+    "srr",
+    "srs",
+    "srt",
+    "strings",
+    "sub",
+    "sup",
+    "sys",
+    "tif",
+    "ttf",
+    "txt",
+    "url",
+    "vob",
+    "website",
+    "wmv",
+    "xml",
+    "xpi",
+)
+
+# combine to one tuple, with unique entries:
+ALL_EXT = tuple(set(POPULAR_EXT + DOWNLOAD_EXT))
+
+
+def has_popular_extension(file_path: str) -> bool:
+    """returns boolean if the extension of file_path is a popular, well-known extension"""
+    file_extension = get_ext(file_path)[1:]  # without the dot
+    return file_extension in ALL_EXT
+
+
+def all_possible_extensions(file_path: str) -> typing.List[str]:
```

Comment:
If you only import the things from typing that you actually need (i.e. `from typing import List`) and the `typing.` part can be dropped. Function definitions tend to be long enough already without extra baggage.